File: examples/simple-pipeline.ts
// In production, use: import { ContextRouter } from '@context-router/sdk';
// For local development, use the built SDK:
import { ContextRouter } from '../packages/sdk-typescript/dist/index.js';

async function researchTopic() {
  return {
    topic: 'Context-efficient agents',
    findings: ['Share selected facts', 'Checkpoint completed work'],
  };
}

async function writeDraft(context: string) {
  return { title: 'Smaller handoffs, clearer agents', contextUsed: context };
}

async function main() {
  const router = await ContextRouter.local();

  try {
    const flow = await router.start('Simple pipeline');
    await flow.set('research', await researchTopic());

    const handoff = await flow.handoff({ keys: ['research'], maxTokens: 120 });
    await flow.set('draft', await writeDraft(handoff.summary));

    await flow.checkpoint('draft-created');
    console.log((await flow.get('draft')).value);
    await flow.complete();
  } finally {
    await router.close();
  }
}

main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});

File: examples/simple-pipeline.py
"""Simple linear workflow example for Context Router Python SDK."""

from __future__ import annotations

import asyncio

from context_router import ContextRouter


async def research_topic() -> dict:
    return {
        "topic": "Context-efficient agents",
        "findings": ["Share selected facts", "Checkpoint completed work"],
    }


async def write_draft(context: str) -> dict:
    return {"title": "Smaller handoffs, clearer agents", "contextUsed": context}


async def main() -> None:
    async with await ContextRouter.local() as router:
        flow = await router.start("Simple pipeline")
        await flow.set("research", await research_topic())

        handoff = await flow.handoff(keys=["research"], max_tokens=120)
        await flow.set("draft", await write_draft(handoff.summary))

        await flow.checkpoint("draft-created")
        draft = await flow.get("draft")
        await flow.complete()
        print(draft.value)


if __name__ == "__main__":
    asyncio.run(main())
